packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -151,6 +164,23 @@ function WorklistDataGrid(props) {
   const defaultColumnMinimumWidth = 60;
   const defaultColumnMaximumWidth = 300;
 
+  const [pinnedColumnOffsets, setPinnedColumnOffsets] = useState([0]);
+  const [pinnedColumnsTotalWidth, setPinnedColumnsTotalWidth] = useState(0);
+
+  const calculateOffsets = () => {
+    let cumulativeOffset = 0;
+    const offsetArray = [0];
+    const totalPinnedColumnsCount = hasSelectableRows ? pinnedColumns.length : pinnedColumns.length - 1;
+
+    dataGridColumns.slice(0, totalPinnedColumnsCount).map((pinnedCol) => {
```

Comment:
Wouldn't it be easier to use a simple for loop in this scenario?  I think that would more readable and efficient than slicing an array just to be able to use the map function..

Reply:
We used this to conform with functional programming patterns. Regarding efficiency, the complexity is O(N) either way so I don't think there's a huge difference. Readability is subjective and dependent upon the project's conventions.